pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -84,7 +84,7 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
 	klog.V(2).Infof("creating vm %s ", vmSpec.Name)
 
 	sshKeyData := vmSpec.SSHKeyData
-	if sshKeyData == "" {
```

Comment:
if sshKey data is given by user this still will get into https://github.com/openshift/cluster-api-provider-azure/pull/140/files#diff-79058a3142fd624b3690ff1dc72c0d2cR124 and break?